connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -391,4 +390,158 @@ private String ownerUrl(String connector) {
 
     }
 
+    public static class ConnectorsAndTasks {
+        private final Collection<String> connectors;
+        private final Collection<ConnectorTaskId> tasks;
+        public static final ConnectorsAndTasks EMPTY =
+                new ConnectorsAndTasks(Collections.emptyList(), Collections.emptyList());
+
+        private ConnectorsAndTasks(Collection<String> connectors, Collection<ConnectorTaskId> tasks) {
+            this.connectors = connectors;
+            this.tasks = tasks;
+        }
+
+        public static ConnectorsAndTasks copy(Collection<String> connectors,
+                                              Collection<ConnectorTaskId> tasks) {
+            return new ConnectorsAndTasks(new ArrayList<>(connectors), new ArrayList<>(tasks));
+        }
+
+        public static ConnectorsAndTasks embed(Collection<String> connectors,
+                                               Collection<ConnectorTaskId> tasks) {
+            return new ConnectorsAndTasks(connectors, tasks);
+        }
+
+        public Collection<String> connectors() {
+            return connectors;
+        }
+
+        public Collection<ConnectorTaskId> tasks() {
+            return tasks;
+        }
+
+        public int size() {
+            return connectors.size() + tasks.size();
+        }
+
+        public boolean isEmpty() {
+            return connectors.isEmpty() && tasks.isEmpty();
+        }
+
+        @Override
+        public String toString() {
+            return "{ connectorIds=" + connectors + ", taskIds=" + tasks + '}';
+        }
+    }
+
+
+    public static class WorkerLoad {
+        private final String worker;
+        private final Collection<String> connectors;
+        private final Collection<ConnectorTaskId> tasks;
+
+        private WorkerLoad(
+                String worker,
+                Collection<String> connectors,
+                Collection<ConnectorTaskId> tasks
+        ) {
+            this.worker = worker;
+            this.connectors = connectors;
+            this.tasks = tasks;
+        }
+
+        public static WorkerLoad copy(
+                String worker,
+                Collection<String> connectors,
+                Collection<ConnectorTaskId> tasks
```

Comment:
I don't think this is used anywhere except for tests. Is that intentional?

Reply:
This class has some overlap with other classes such as `ConnectorsAndTasks` and even `LeaderState`. I agree and I would also like to consolidate, but maybe when we are near end of the PR review.